src/cunumeric/index/advanced_indexing_omp.cc

```diff
@@ -98,7 +74,8 @@ struct AdvancedIndexingImplBody<VariantKind::OMP, CODE, DIM, OUT_TYPE> {
 
     const auto max_threads = omp_get_max_threads();
     const size_t volume    = rect.volume();
-    ThreadLocalStorage<int64_t> offsets(max_threads);
+    auto kind    = CuNumeric::has_numamem ? Memory::Kind::SOCKET_MEM : Memory::Kind::SYSTEM_MEM;
+    auto offsets = create_buffer<int64_t>(max_threads, kind);
```

Comment:
Was there an issue with `offsets` being a `ThreadLocalStorage` object, after you removed the thrust calls?

Reply:
Nothing is wrong with it since I don't use Thrust anymore. Replacing buffer with ThreadLocalStorage